tests/check_new_syntax.py

```diff
@@ -63,30 +83,49 @@ def visit_Subscript(self, node: ast.Subscript) -> None:
     class AnnotationFinder(ast.NodeVisitor):
         def __init__(self) -> None:
             self.set_from_collections_abc = False
+            self.context_manager_from_typing = False
 
         def visit_ImportFrom(self, node: ast.ImportFrom) -> None:
             if node.module == "collections.abc":
                 imported_classes = node.names
                 if any(cls.name == "Set" for cls in imported_classes):
                     self.set_from_collections_abc = True
+            if node.module == "typing":
+                imported_classes = node.names
+                if any(cls.name == "ContextManager" for cls in imported_classes):
+                    self.context_manager_from_typing = True
```

Comment:
Why is there not similar code for `AsyncContextManager`?

Also, there's no contextlib.AbstractAsyncContextManager in Python 3.6, but that doesn't seem to be a problem now, and likely won't become a problem as we won't support Python 3.6 for much longer.

Reply:
Purely because there is a third-party stub in typeshed (`humanfriendly`) that has a class called `ContextManager`, thus necessitating this block of code; but there is no third-party stub file in typeshed that has a class named `AsyncContextManager`. I figured that if a stub ever were added to typeshed that had a class named `AsyncContextManager, we could add a block of code like this then; but it's not as if we check to see whether `Tuple` has been imported from `typing` before raising an error, because it just isn't really an issue 🙂

Reply:
If a third-party library has a class named `Tuple`, I expect it to cause some kind of problems, but at first sight, a class named `AsyncContextManager` doesn't seem very likely to conflict with anything. So I think we should check for that too, just to prevent having a wtf moment some time in the future.

Reply:
Fair, I'll add it in

Reply:
I did have a `y tho` moment when I saw `humanfriendly` had a class named `ContextManager`